src/utils/utils.ts

```diff
@@ -47,3 +49,57 @@ export const renderTags = (tags: string[], isMobileDevice: boolean) => {
 export const isNotEmpty = (value: string) => {
   return value.trim() !== "";
 };
+
+export const calculateSourcesChart = (articles: Article[]) => {
```

Comment:
Greate usage of utils file. if it becomes too large you can consider splitting it into topics